### DISCUSSION TITLE
Issues in GND computation for plagioclase

### DISCUSSION THREAD

**Discussion by MaelAllard** [mtex-toolbox/mtex#511] [Ask Anything]
Hi MTex community,

I'm currently dealing with an issue while trying to compute GND in plagioclase.
The problem occurs when defining slip systems, from TEM observations it is observed that the dislocation line
of [100] (001) is in (001) between [1-10] and [2-10] axes (Montardi and Mainprice, 1987). The problem is these axes are respectively at 57.29° and 38.15° from the Burger vector [100], and the code returns the following error (example for the first one):

`% Define Burgers and dislocation lines`
`B_100 = Miller(1,0,0,CS,'uvw');`
`L_1m10 = Miller(1,-1,0,CS,'uvw');`
`% Set the energies of dislocations`
`E_edge_100 = (norm(ebsd_smoothed.CS.aAxis))^2; % smoothing (denoising) of ebsd data with the halfQuadraticFilter`
`% Create dislocation systems`
`dS_001_100a = dislocationSystem(B_100, L_1m10,E_edge_100)`

` >>`
`Error using dislocationSystem`
`line vector and burgers vector should be either be orthogonal! or identical`

So it seems that it is forced to be higher to certain angle in "dislocationSystem" (works for angles of 64°),
but for triclinic symmetries may it be allowed to have lines at lower angles ?
There is also another point which is not clear for me, that when defining a screw dislocation system no glide plane
can be specify but some are observed on different planes (i.e. for different slip systems). Is this none specification normal
or is it a limitation of the method ?
![100 (001)_slip_syst](https://user-images.githubusercontent.com/57705108/91571297-9b420880-e946-11ea-8e42-cf1e8be0ee6c.jpg)

Thank you,

Maël Allard

**Comment by kilir** [comment]
Hi Mael,
I was also wondering what those actually are - mixed mode dislocations from the point of view that edge dislocations have the dislocation line normal to the burgers vector? If you comment line 70 to overcome the warning, however, at the moment I'm not really sure if this brings you closer to your GND. 
What dislocation energy would you assume?
Cheers,
Rüdiger

**Comment by MaelAllard** [comment]
Hi Rüdiger,
In the literature, it seems that it is described as an edge dislocation, even if the "true edge" character of dislocations is not possible in triclinic because the angle is usually different from 90°. For example, it does work in MTEX with [001] (010) slip system where [100] which is the line is at 116° (or 64°) from the Burgers. I did the calculation as if it was an edge dislocation without problems is this case.
After commenting line 70 it works, but what could be the problem with the resulting GNDs ?
For the moment, I define the dislocation energy as the one of an edge dislocation.

Cheers,
Maël

**Comment by kilir** [comment]
The fact, that Mtex did not complain about the 116° is unintentional since it checks for the line being parallel or perpendicular.
The curvature associated with those dislocations is different, hence one could expect a different result.
However, in that case you would also need to give the "proper" dislocation energy since this is the thing being minimized during GND calculation. 

You could set up an synthetic model to test the differences.



```
% oblique
CS = crystalSymmetry('1',[8.5 13 7],[90.3 116 89]*degree)
B = Miller(1,0,0,CS,'uvw');
L1 = Miller(1,-1,0,CS,'uvw');
dS1= dislocationSystem(B,L1)
dT1 = curvature(dS.tensor);
angle(dS1.b,dS1.l,'noSymmetry')/degree

% edge 
L2 = Miller(0,-1,0,CS,'hkl');
dS2= dislocationSystem(B, L2)
dT2 = curvature(dS2.tensor);
angle(dS2.b,dS2.l,'noSymmetry')/degree


% screw 
L3 = Miller(1,0,0,CS,'uvw');
% Create dislocation systems
dS3= dislocationSystem(B, L3)
dT3 = curvature(dS3.tensor);
angle(dS3.b,dS3.l,'noSymmetry')/degree


plot(dT1,'contourf')
mtexTitle('dT1')
nextAxis
plot(dT2,'contourf')
mtexTitle('dT2')
nextAxis
plot(dT3,'contourf')
mtexTitle('dT3')
```

**Comment by MaelAllard** [comment]
Hi Rüdiger,
Thank you for your answer.

For now to set energies of dislocations I use the following formulas:
`Edge = burgers^2`
`Screw = burgers^2 / (1 - Poisson ratio) `

are they still applicable here?
Also, what is the way to set up oblique dislocations energies? 
Because summing up Edge and Screw energies and then divided the result by 2 may not be realistic. In the case of plagioclase, it is very common to have rectangular loops of dislocations with elongated screw segments, so it may indicate higher energy for edge segments.

**Comment by kilir** [comment]
Hi Mael,
from my point of view - which is clearly debatable - all of those are not realistic (up to a certain point).
a) usually (e.g. see Hull & Bacon) crystals are assumed to be elastically isotropic - which is usually not the case
b) the computation for elastically anisotropic media is given for example in Hirth & Lothe p.443ff (chapter 13-3 Straight dislocations in anisotropic media / Stresses and Energies) and there may not  be a solution for arbitrary dislocations (there are some numerical solution from Heinisch et al 1975 for some phases)
c) this is all about single, isolated dislocations out there in the wild. Those structures we usually see above noise in EBSD data are low angle boundaries and we bravely identify them as subgrain boundaries/ dislocations walls. I think we do not know too much about the dislocation configuration inside those structures. Hence it's somewhat of a big leap to take elastic energies for isolated dislocations in isotropic media and use those for what we see in EBSD data.

Cheers,
Rüdiger

**Comment by MaelAllard** [comment]
Hi Rüdiger,
Thank you for these informations,
I will go and see in these articles.

Cheers,
Maël
